jen/examples_test.go

```diff
@@ -1295,6 +1294,19 @@ func ExampleTag() {
 	// }
 }
 
+func ExampleTag_withQuotesAndNewline() {
+	c := Type().Id("foo").Struct(
+		Id("A").String().Tag(map[string]string{"json": "a"}),
+		Id("B").Int().Tag(map[string]string{"json": "b", "bar": "the value of\nthe\"bar\" tag"}),
+	)
+	fmt.Printf("%#v", c)
+	// Output:
+	// type foo struct {
+	// 	A string `json:"a"`
+	// 	B int    `bar:"the value of\nthe \"bar\" tag" json:"b"`
```

Comment:
Previously, we would get `B int    "bar:\"the value of\nthe\"bar\" tag\" json:\"b\""`

Notice that the quotes around `bar` are escaped at the same level as the quotes for the tag value, meaning the [`reflect.StructTag.Get` parser](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/reflect/type.go;l=1207) will recognize the first quote after opening as the end of the value.